src/PLLModelGetLanguagesListDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,133 @@
+ */
+
+declare(strict_types=1);
+
+namespace WPSyntex\Polylang\PHPStan;
+
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use PLL_Language;
+use PLL_Model;
+
+class PLLModelGetLanguagesListDynamicMethodReturnTypeExtension implements DynamicMethodReturnTypeExtension {
+	public function getClass(): string {
+		return PLL_Model::class;
+	}
+
+	public function isMethodSupported( MethodReflection $methodReflection ): bool {
+		return $methodReflection->getName() === 'get_languages_list';
+	}
+
+	public function getTypeFromMethodCall( MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope ): Type {
+		$args = $methodCall->getArgs();
+
+		if ( count( $args ) === 0 ) {
+			// Called without arguments.
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		$argumentType = $scope->getType( $args[0]->value );
+
+		if ( ! $argumentType->isArray()->yes() ) {
+			// Called with an argument that is not an array-like.
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		if ( $argumentType instanceof IntersectionType && $argumentType->isIterable() ) {
+			$fieldsInst = new ConstantStringType( 'fields' );
+
+			foreach( $argumentType->getTypes() as $type ) {
+				if ( ! $type->hasOffsetValueType( $fieldsInst )->yes() ) {
+					continue;
+				}
+
+				$fieldsType = $type->getOffsetValueType( $fieldsInst );
+
+				if ( ! $fieldsType instanceof ConstantStringType ) {
+					// The 'field' argument is not a string.
+					return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+				}
+
+				$fieldsValue = $fieldsType->getValue();
+				break;
+			}
+		}
+
+		if ( ! isset( $fieldsValue ) && $argumentType instanceof ArrayType ) {
+			$argumentKeys = $argumentType->getKeysArray();
+
+			if ( $argumentKeys instanceof ConstantArrayType ) {
+				foreach ( $argumentKeys->getValueTypes() as $index => $key ) {
+					if ( $key->getValue() !== 'fields' ) {
+						continue;
+					}
+
+					$fieldsType = $argumentType->getValuesArray()->getValueTypes()[ $index ];
+
+					if ( ! $fieldsType instanceof ConstantStringType ) {
+						// The 'field' argument is not a string.
+						return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+					}
+
+					$fieldsValue = $fieldsType->getValue();
+					break;
+				}
+			}
+		}
+
+		if ( empty( $fieldsValue ) ) {
+			// Without 'fields' argument, or empty value.
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
```

Comment:
```suggestion
			return new ArrayType( new IntegerType(), new MixedType() );
```
Here we were not able to find the `fields` key, but it doesn't mean it doesn't exist IMHO.
See [here](https://github.com/polylang/polylang-phpstan/blob/languagelist-dynamic-return-type-tests/tests/data/get_languages_list.php#L19), we pass an array with unknown content.

Reply:
Done in [56cbfef](https://github.com/polylang/polylang-phpstan/pull/3/commits/56cbfef1bda925922cea92b3254887212f64d0a8).